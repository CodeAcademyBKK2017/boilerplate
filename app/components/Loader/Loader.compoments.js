import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './Loader.compoments.style';
import {ActivityIndicator, Modal, View
} from 'react-native';

export default class Loader extends Component {
  render () {
    return (
      <Modal transparent={true} visible={this.props.show} style={[styles.container, styles.horizontal]}>
        <View  style={styles.containerView}>
          <ActivityIndicator animating={this.props.show} size='large' color='#0000ff' />
        </View>
      </Modal>
    );
  }
}

Loader.propTypes = {
  show: PropTypes.bool
};
  