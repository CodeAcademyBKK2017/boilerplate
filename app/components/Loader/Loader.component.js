import Overlay from 'react-native-modal-overlay';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './Loader.style';
import {ActivityIndicator} from 'react-native';

export default class Loader extends Component {
  render () {
    const showHide = this.props.visibility;
    return (
      <Overlay visible={showHide.isVisible}
        closeOnTouchOutside={true}  
        animationType='slideInLeft'
        animationDuration={1000} 
        containerStyle={styles.containerModal}
        childrenWrapperStyle={styles.childrenModal}>
        <ActivityIndicator size='large' color='#32ff00' />
      </Overlay>
    );
  }
}
Loader.propTypes = {
  visibility: PropTypes.object
};
  