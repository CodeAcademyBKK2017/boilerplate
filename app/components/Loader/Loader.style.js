import {
  StyleSheet
} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#00000033'
  },
  horizontal: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    padding: 10
  }
});

export default styles;
  