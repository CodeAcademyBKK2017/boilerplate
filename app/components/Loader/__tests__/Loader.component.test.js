import 'react-native';
import Loader from '../Loader.component';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Loader', () => {
  it('snapshot test', () => {
    const param = {Isvisible: false};
    const snapshot = renderer.create(<Loader visibility={param} />);
    expect(snapshot).toMatchSnapshot();
  });
});