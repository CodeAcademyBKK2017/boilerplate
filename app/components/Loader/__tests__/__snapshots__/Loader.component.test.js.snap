// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader snapshot test 1`] = `
<Modal
  animationOutType="fadeOut"
  animationType="none"
  hardwareAccelerated={false}
  onClose={[Function]}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(37, 8, 10, 0.78)",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0,
        "padding": 20,
      }
    }
    testID={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "padding": 20,
          },
          Object {
            "backgroundColor": "transparent",
          },
        ]
      }
      testID={undefined}
    >
      <ActivityIndicator
        animating={true}
        color="#32ff00"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </View>
</Modal>
`;
